feat: introduce WAL archive volume (#513)
cloudnative-pg/cloudnative-pg: docs/src/storage.md

      volumeMode: Filesystem
```

## Volume for WAL

> By default, PostgreSQL stores all its data in the so called `PGDATA`.

Reviewer: ```suggestion
By default, PostgreSQL stores all its data in the so-called `PGDATA` (a directory).
```

> One of the core directories inside `PGDATA` is called `pg_wal` (historically

Reviewer: ```suggestion
One of the core directories inside `PGDATA` is `pg_wal` (historically
```

known as `pg_xlog` in PostgreSQL), which contains the log of transactional
changes occurred in the database, in the form of segment files.

!!! Info

>     Normally, each segment is 16 MB in size, but the size is can be changed

Reviewer: ```suggestion
    Normally, each segment is 16 MB in size, but the size can be configured
```

>     at cluster initialization time through the `walSegmentSize` option, as

Reviewer: ```suggestion
    through the `walSegmentSize` option, applied at cluster initialization time, as
```

    described in ["Bootstrap an empty cluster"](bootstrap.md#bootstrap-an-empty-cluster-initdb).

> While in most of the cases having `pg_wal` on the same volume where `PGDATA`

Reviewer: ```suggestion
While in most cases, having `pg_wal` on the same volume where `PGDATA`
```

NAME                           READY   STATUS      RESTARTS   AGE
cluster-example-1              1/1     Running     0          5m58s
cluster-example-2              1/1     Running     0          5m43s

> cluster-example-4-join-v2      0/1     Completed   0          17s

Author: maek checks throws error for the 'bfg' keyword

$ kubectl delete pvc/cluster-example-3 pod/cluster-example-3
```

!!! Important

>     In case you have instanciated a dedicated WAL volume, both PCVs will have to be deleted during this process.

Reviewer 2: ```suggestion
    In case you have instantiated a dedicated WAL volume, both PCVs will have to be deleted during this process.
```

Reviewer 3: Fixed

>     In case you have instantiated a dedicated WAL volume, both PCVs will have to be deleted during this process.

Reviewer: ```suggestion
    In case you have created a dedicated WAL volume, both PVCs will have to be deleted during this process.
```
